### ISSUE TITLE
ForceStrictCondition should not affect individual array elements.

### ISSUE THREAD

**Issue by Abhishek-kumar09** [checkstyle/checkstyle#9086]
Extended from https://github.com/checkstyle/checkstyle/pull/8083#discussion_r534196175

javac Test.java 
```
No output
```

$ cat Test.java
```java
@InputIndentationAnnArrInit2.Foo({ //indent:0 exp:0
            @InputIndentationAnnArrInit2.Bar, //indent:12 exp:>=2 // should not allow this indentation.
            // because the element should only follow arrayInit, LineWrapping and basicOffset
            //  and no other except that
            // moreover, currently it is also following forceStrictCondition, due to which
            // the results/checkstyle output is changed with 
            // changing value of forceStrictCondition which should not be, because
            // it is not a lineWrapped element but an array child.
}) //indent:0 exp:0

// helper code
class InputIndentationAnnArrInit2 { //indent:0 exp:0
  @interface Foo { //indent:2 exp:2
    Bar[] value() default {}; //indent:4 exp:4

    String[] values() default {"Hello", "Checkstyle"}; //indent:4 exp:4
  } //indent:2 exp:2

  @interface Bar {} //indent:2 exp:2
} //indent:0 exp:0
```

$ cat config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    <module name="Indentation">
      <property name="arrayInitIndent" value="2" />
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="8"/>
    </module>
    </module>
</module>
```

# Actual Behaviour
$ java -jar checkstyle-8.38-all.jar -c config.xml Test.java 
```
Starting audit...
Audit done.
```

# Expected Behaviour
Error on line 2
```
Starting audit...
[ERROR] /Test.java:2:13: 'annotation array initialization' child has incorrect indentation level 12,
          expected level should be one of the following: 2, 8, 35, 41. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

<details><summary>Verbal</summary>
<p>

**Concise description:**
Currently after merging PR #8083, annotation array initialization is supporting forceStrictCondition false and true properties.
Though the element indentation is made more flexible now, but it is somewhere not relatable to documentation of checkstyle upon array init indentation. 
The array elements should follow arrayinit Indentation and thus it should not care by forceStrictCondition.
because according to documentation: 
> Force strict indent level in line wrapping case. If value is true, line wrap indent have to be same as lineWrappingIndentation parameter. If value is false, line wrap indent could be bigger on any value user would like.

ForceStrictCondition should only be useful with lineWrapping.

**Describe the solution**
remove support for flexible indent levels for array children, they should follow pattern before merging PR #8038 for arrayInitializations.

**A doubt**
In array Initilizations we were following arrayInit, basicOffset, and linewrapping, so that user could get some flexibilty with their config, or for making the placements of children less confusing. But since we were using lineWrapping there, can't we have it benefits of forceStrictCondition false and  any indent level after minimum requirement.

</p>
</details>

**Comment by romani** [comment]
@Abhishek-kumar09, please follow issue template

**Comment by Abhishek-kumar09** [comment]
@romani is it good. 
I followed this template: https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE/feature_request.md

**Comment by romani** [comment]
@Abhishek-kumar09 , please use https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F
it is simple bug that we did not noticed before release, but released and users did not upgrade yet to latest version to start complaining.

**Comment by Abhishek-kumar09** [comment]
@romani updated issue decription :)

**Comment by strkkk** [comment]
Fix is merged
